### DISCUSSION TITLE
How to use -filelist

### DISCUSSION THREAD

**Discussion by Surxe** [SteamRE/DepotDownloader#552] [Q&A] [answered]
Hi! I got DepotDownloader working with
`./DepotDownloader -app 1422450 -depot 1422456 -manifest 7423908266886961121 -username <myuser> -password <mypass> -remember-password`

but I am having some trouble using the `-filelist` parameter. I specifically want to download the `<steam_path>steamapps\common\Deadlock\game\citadel\steam.inf`.

I verified the `steam.inf` files exists in this manifest first by downloading the whole manifest with the first command. I tried `-filelist game/citadel/steam.inf` and `-filelist game\citadel\steam.inf` to no luck. The former displays this error
![image](https://github.com/user-attachments/assets/ea795bdd-9a97-4637-bfa2-ab5614c38e71)

I installed the windows x64 package from the releases area.

Any help would be greatly appreciated, thank you!

**Comment by yaakov-h** [comment]
filelist points to a file which contains the list of items to download. it is not itself the list of files to download.

**Reply by Surxe** [reply]
> filelist points to a file which contains the list of items to download. it is not itself the list of files to download.

Hi, could you elaborate? I'm still a bit confused, I don't understand the difference here. Thanks!

**Reply by Surxe** [reply]
Is it like, I create a file list.txt, and inside the file I put "game/citadel/steam.inf"? Whats the syntax for adding multiple files into list.txt?

**Reply by psychonic** [reply, marked as answer]
> Is it like, I create a file list.txt, and inside the file I put "game/citadel/steam.inf"?

Yes.

> Whats the syntax for adding multiple files into list.txt?

One file pattern per line

**Reply by Surxe** [reply]
Perfect thank you! Also, is there a doc you can refer me to for downloading all manifests from a depo? If I omit the manifest, it seems to be using the latest manifest. Thank you!

**Reply by psychonic** [reply]
There is no built-in way for downloading all manifests for a depot. Steam doesn't expose that information. It only gives the current, latest manifest id for a given branch.

**Reply by Surxe** [reply]
Noted. What do you recommend for querying all manifest id's for a depot so I can write some quick python to do it? Thank you in advance!

**Reply by psychonic** [reply]
I don't know of any source that's available to query for that. Steam does not provide that information. If you have your own service, you could track, store, and serve it over time.

**Reply by Surxe** [reply]
Got it, thank you again for your help!

**Reply by Surxe** [reply]
Got a very quick python script to do it for me for every manifest, if anyone is interested in the future feel free to @ me :)
